## Collecte de données
Le but du projet étant de mettre en place un modèle de prédictions des résultats de match de foot, nous allons dans un premier temps collecter les résultats des matchs des 6 dernières saisons de Premier League (Championnat Anglais)

In [13]:
import pandas as pd

seasons = {
    "2018-2019": "1819",
    "2019-2020": "1920",
    "2020-2021": "2021",
    "2021-2022": "2122",
    "2022-2023": "2223",
    "2023-2024": "2324"
}

all_matches = []

for season, code in seasons.items():
    url = f"https://www.football-data.co.uk/mmz4281/{code}/E0.csv"
    print(f"Downloading {season}")

    df = pd.read_csv(url)
    df["season"] = season
    all_matches.append(df)

matches_df = pd.concat(all_matches, ignore_index=True)

matches_df.to_csv(
    "../data/raw/premier_league_matches_raw.csv",
    index=False
)

matches_df.head()


Div        Date      HomeTeam        AwayTeam  FTHG  FTAG FTR  HTHG  HTAG  \
0  E0  10/08/2018    Man United       Leicester     2     1   H     1     0   
1  E0  11/08/2018   Bournemouth         Cardiff     2     0   H     1     0   
2  E0  11/08/2018        Fulham  Crystal Palace     0     2   A     0     1   
3  E0  11/08/2018  Huddersfield         Chelsea     0     3   A     0     2   
4  E0  11/08/2018     Newcastle       Tottenham     1     2   A     1     2   

  HTR  ... AvgC<2.5  AHCh  B365CAHH  B365CAHA  PCAHH  PCAHA  MaxCAHH  MaxCAHA  \
0   H  ...      NaN   NaN       NaN       NaN    NaN    NaN      NaN      NaN   
1   H  ...      NaN   NaN       NaN       NaN    NaN    NaN      NaN      NaN   
2   A  ...      NaN   NaN       NaN       NaN    NaN    NaN      NaN      NaN   
3   A  ...      NaN   NaN       NaN       NaN    NaN    NaN      NaN      NaN   
4   A  ...      NaN   NaN       NaN       NaN    NaN    NaN      NaN      NaN   

   AvgCAHH  AvgCAHA  
0      NaN      NaN  
1      NaN      NaN  
2      NaN      NaN  
3      NaN      NaN  
4      NaN      NaN  

[5 rows x 125 columns]

In [14]:
len(matches_df)

2280

nous avons 380 matchs par saison en premier league, donc pour les 6 saisons nous aurons 2280 matchs. ce qui conclue la première étape du projet.